extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/ProgrammaticApiTest.java

```diff
@@ -203,6 +214,47 @@ public void testExceptionInValueLoader() throws Exception {
         assertGetIfPresentMissingKey(key);
     }
 
+    @Test
+    public void testCompute() {
+        try {
+            String value = cacheCompute.compute(KEY_COMPUTE, (key, previousValue) -> {
+                assertEquals(KEY_COMPUTE, key);
+                assertNull(previousValue);
+                return VALUE_COMPUTE;
+            });
+            assertEquals(VALUE_COMPUTE, value);
+            assertEquals(VALUE_COMPUTE, cacheCompute.get(KEY_COMPUTE, Function.identity()).await().atMost(Duration.ofSeconds(1)));
+        } finally {
+            cache.invalidate(KEY_COMPUTE);
+        }
+    }
+
+    @Test
+    public void testComputeConcurrent() {
+
+        String CONCURRENT_VALUE = "CONCURRENT_VALUE";
+
+        try {
+            assertThrows(RuntimeException.class, () -> cacheCompute.compute(KEY_COMPUTE, (key, previousValue) -> {
+                CountDownLatch latch = new CountDownLatch(1);
+                new Thread(() -> {
+                    cache.compute(KEY_COMPUTE, (k, p) -> CONCURRENT_VALUE);
+                    latch.countDown();
+                }).start();
+                try {
+                    latch.await();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+                return VALUE_COMPUTE;
+            }));
+            assertEquals(CONCURRENT_VALUE, cacheCompute.get(KEY_COMPUTE, Function.identity()).await().atMost(Duration.ofSeconds(1)));
+        } finally {
+            cache.invalidate(KEY_COMPUTE);
+        }
+    }
```

Comment:
@ben-manes this is the failing test. I am trying to insert concurrently an element in the cache to make the compute method fail because of a concurrent insertion, however, it is not throwing.
With the debugger, I saw that putIfAbsent return null even if the CONCURRENT_VALUE has been inserted in the meantime

Reply:
`CONCURRENT_VALUE` is being inserted into `cache` and the check in on a different instance,
`assertEquals(CONCURRENT_VALUE, cacheCompute.get(KEY_COMPUTE, identity())`

Are you sure `putIfAbsent` was on `cache` and not `cacheCompute`?